src/Kestrel.Core/ListenOptions.cs

```diff
@@ -126,6 +129,10 @@ public FileHandleType HandleType
         /// </remarks>
         public List<IConnectionAdapter> ConnectionAdapters { get; } = new List<IConnectionAdapter>();
 
+        public IServiceProvider ApplicationServices => KestrelServerOptions?.ApplicationServices;
```

Comment:
`ApplicationServices` is part of the interface. It will be used to activate middleware, just like `IApplicationBuilder. ApplicationServices`